runners/core-java/src/test/java/org/apache/beam/runners/core/metrics/MetricsContainerImplTest.java

```diff
@@ -182,6 +182,38 @@ public void testMonitoringInfosArePopulatedForUserCounters() {
     assertThat(actualMonitoringInfos, containsInAnyOrder(builder1.build(), builder2.build()));
   }
 
+  @Test
+  public void testMonitoringInfosArePopulatedForUserDistributions() {
+    MetricsContainerImpl testObject = new MetricsContainerImpl("step1");
+    DistributionCell c1 = testObject.getDistribution(MetricName.named("ns", "name1"));
+    DistributionCell c2 = testObject.getDistribution(MetricName.named("ns", "name2"));
```

Comment:
I see the old-style metric names. Are you also adding a unit test for the MonitoringInfoMetricName?

Reply:
Okay, I added a test for this case as well. But this required adding the MonitoringInfoSpec for SampledByteCount, which I was going to do in this PR which is queued up after this one. But I have pulled it into this PR.

https://github.com/apache/beam/pull/8416/files